src/test/java/com/amazon/opendistroforelasticsearch/sql/rewriter/alias/TableAliasPrefixRemoveRuleTest.java

```diff
@@ -0,0 +1,141 @@
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.rewriter.alias;
+
+import com.alibaba.druid.sql.SQLUtils;
+import com.alibaba.druid.sql.ast.expr.SQLQueryExpr;
+import com.amazon.opendistroforelasticsearch.sql.util.SqlParserUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test cases for field name prefix remove rule.
+ */
+public class TableAliasPrefixRemoveRuleTest {
+
+    @Test
+    public void queryWithUnAliasedTableNameShouldMatch() {
+        query("SELECT account.age FROM accounts").shouldMatchRule();
+    }
+
+    @Test
+    public void queryWithUnAliasedTableNameInSubQueryShouldNotMatch() {
+        query("SELECT * FROM test t WHERE t.name IN (SELECT accounts.name FROM accounts)").shouldNotMatchRule();
+    }
+
+    @Test
+    public void queryWithoutUnAliasedTableNameShouldMatch() {
+        query("SELECT a.age FROM accounts a WHERE a.balance > 1000").shouldMatchRule();
+    }
+
+    @Test
+    public void joinQueryWithoutUnAliasedTableNameShouldNotMatch() {
+        query("SELECT * FROM accounts a1 JOIN accounts a2 ON a1.city = a2.city").shouldNotMatchRule();
+    }
+
+    @Test
+    public void nestedFieldQueryWithoutUnAliasedTableNameShouldNotMatch() {
+        query("SELECT * FROM accounts a, a.project p").shouldNotMatchRule();
+    }
+
+    @Test
+    public void selectedFieldNamePrefixedByUnAliasedTableNameShouldRemoveTableNamePrefix() {
+        query("SELECT accounts.age FROM accounts").shouldBeAfterRewrite("SELECT age FROM accounts");
+        query("SELECT accounts.age FROM accounts/temp").shouldBeAfterRewrite("SELECT age FROM accounts/temp");
+    }
+
+    @Test
+    public void allFieldNamePrefixedByUnAliasedTableNameEverywhereShouldRemoveTableNamePrefix() {
+        query(
+            "SELECT accounts.age, AVG(accounts.salary) FROM accounts WHERE accounts.age > 10 " +
+            "GROUP BY accounts.age HAVING AVG(accounts.balance) > 1000 ORDER BY accounts.age"
+        ).shouldBeAfterRewrite(
+            "SELECT age, AVG(salary) FROM accounts WHERE age > 10 " +
+            "GROUP BY age HAVING AVG(balance) > 1000 ORDER BY age"
+        );
+    }
+
+    @Test
+    public void selectedFieldNamePrefixedByTableAliasShouldRemoveTableAliasPrefix() {
+        query("SELECT a.age FROM accounts a").shouldBeAfterRewrite("SELECT age FROM accounts");
+        query("SELECT a.age FROM accounts/temp a").shouldBeAfterRewrite("SELECT age FROM accounts/temp");
```

Comment:
Can we add one more `query("SELECT age FROM accounts/temp a").shouldBeAfterRewrite("SELECT age FROM accounts/temp");` , here age does not have table alias.

Reply:
Sure, will do.

Reply:
I think it has been check here `unAliasedFieldNameShouldNotBeChanged()`, you can skip this.